### PR TITLE
Create API routes for creating and editing projects

### LINKED ISSUE #11: Create API routes for creating and editing projects
Called by: 
- #52

Functionality: 
- Insert or edits a personal project or a project of an organization, in which you are an administrator, in the projects table

Restricted to:
- logged in users
- if they are creating a project of an organization: admins of that organization

### PR DESCRIPTION
**Pull request by ddhanesha**
Resolves #11 

I have created two APIs to create organizational projects and personal projects. I will add testing steps once its ready for review.

### CODE REVIEW

**Comment by ddhanesha** [comment]
This PR is ready for review. @positiveimpact there are lot of changes here. So if you want we can setup a call to go over all the changes.

File: backend/organization/utility/project.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    if 'country' in data:
+        project.country = data['country']
+    if 'city' in data:
+        project.city = data['city']
+    if 'image' in data:
+        project.image = data['image']
+    if 'description' in data:
+        project.description = data['description']
+
+    project.url_slug = project.name + str(project.id)
```
Comment (positiveimpact) [inline backend/organization/utility/project.py:49]: We could consider making easier to remember urls in the long run, so people can also tell somebody "Just go to climateconnect.earth/xyz to view our project". But in the short run I think this is fine :)

File: backend/organization/views/project_views.py [hunk 2]
```diff
@@ -23,14 +37,102 @@ def get_serializer_class(self):
...
-    serializer_class = ProjectSerializer
-    pagination_class = None
+"""
+Note: I am splitting organization project creation and personal project creation into
+two separate APIs. It makes more sense do two APIs so that we can manage user accessibility
+by project or organization member roles.
+"""
+
+
+class OrganizationProjectsView(APIView):
```
Comment (positiveimpact) [inline backend/organization/views/project_views.py:47]: Shouldn't this be called something like `OrganizationsCreateProjectView`? I think the name doesn't really make it clear that this view is to create a project.

File: backend/organization/views/project_views.py [hunk 3]
```diff
@@ -23,14 +37,102 @@ def get_serializer_class(self):
...
+        project_creation_pre_checks(request.data)
+        project = create_new_project(request.data)
+
+        if 'order' in request.data:
+            order = request.data['order']
+            ProjectParents.objects.create(
+                project=project, parent_organization=organization,
+                parent_user=request.user, order=order
+            )
+
```
Comment (positiveimpact) [inline backend/organization/views/project_views.py:65]: In the "share project" interfaces you can already add your team members. Should it be possible to pass team members to this API and have them be created by this view like the projectparents or should the frontend do an additional request for that? I would vote for the first option. The same thing applies to the personal project create API.

File: backend/organization/views/project_views.py [hunk 4]
```diff
@@ -23,14 +37,102 @@ def get_serializer_class(self):
...
+        project = create_new_project(request.data)
+        if 'order' in request.data:
+            order = request.data['order']
+            ProjectParents.objects.create(
+                project=project, parent_user=request.user, order=order
+            )
+
+        # Here a user is creating an individual project so we will be adding them as a project member
+        role = Role.objects.get(name="Project Administrator")
+        ProjectMember.objects.create(
```
Comment (positiveimpact) [inline backend/organization/views/project_views.py:84]: Here we could directly add the other members, that the user specified in the "share project" interace.

File: backend/organization/views/project_views.py [hunk 5]
```diff
@@ -23,14 +37,102 @@ def get_serializer_class(self):
...
+            project=project, user=request.user, role=role
+        )
+        serializer = ProjectSerializer(project)
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+
+class ProjectAPIView(APIView):
+    permission_classes = [OrganizationProjectCreationPermission]
```
Comment (positiveimpact) [inline backend/organization/views/project_views.py:92]: Does setting the permission to this mean that users who don't fulfil this permission also can't do the get request to view a project?

File: backend/organization/admin.py [hunk 6]
```diff
@@ -43,4 +44,11 @@ class OrganizationMemberAdmin(admin.ModelAdmin):
     list_filter = ('role__name',)
 
 
-admin.site.register(OrganizationMember, OrganizationMemberAdmin)
\ No newline at end of file
+admin.site.register(OrganizationMember, OrganizationMemberAdmin)
+
+
+class ProjectParentsAdmin(admin.ModelAdmin):
+    search_fields = ('project__name', 'organization__name')
```
[the remark this answers is not in the document]
Comment (ddhanesha) [reply backend/organization/admin.py:51]: Oh this is on django admin. Its for internal use only. For us and whoever else can access django admin.

File: backend/organization/utility/project.py (hunk 1 above)
Comment (ddhanesha) [reply backend/organization/utility/project.py:49]: Yeah agreed. For now I think this works.

File: backend/organization/views/project_views.py (hunk 2 above)
Comment (ddhanesha) [reply backend/organization/views/project_views.py:47]: Good call. Yeah I will change the name

File: backend/organization/views/project_views.py (hunk 3 above)
Comment (ddhanesha) [reply backend/organization/views/project_views.py:65]: Yeah I think with create API we can add project members in the same API. For editing each project we should create a separate API to add project members. Hope that makes sense.

File: backend/organization/views/project_views.py (hunk 4 above)
Comment (ddhanesha) [reply backend/organization/views/project_views.py:84]: For create API will do that.

File: backend/organization/views/project_views.py (hunk 5 above)
Comment (ddhanesha) [reply backend/organization/views/project_views.py:92]: Anyone who is authenticated can read a project. Only people with the permission will be able to edit project.

**Comment by ddhanesha** [comment]
@positiveimpact this PR is ready for a second review. If everything looks good we can merge the PR.

**Comment by positiveimpact** [comment]
Looks good to me! Merging.

**Pull request merged**